homeassistant/auth/auth_store.py

```diff
@@ -295,25 +250,20 @@ def async_log_refresh_token_usage(
         refresh_token.last_used_ip = remote_ip
         self._async_schedule_save()
 
-    async def _async_load(self) -> None:
+    async def async_load(self) -> None:
         """Load the users."""
-        async with self._lock:
-            if self._users is not None:
-                return
-            await self._async_load_task()
+        # Make sure that we're not overriding data if 2 loads happened at the
+        # same time
+        if self._loaded:
+            return
```

Comment:
Shouldn't we just blow up here? If it is happening, we might now return too early before auth is actually loaded. This shouldn't happen inside HA.

Reply:
159edf48380a65055de50d35220476e5e2130dd2